google/auth/compute_engine/_metadata.py

```diff
@@ -133,12 +137,17 @@ def get(request, path, root=_METADATA_ROOT, recursive=False, retry_count=5):
             retrieving metadata.
     """
     base_url = urlparse.urljoin(root, path)
-    query_params = {}
 
     if recursive:
-        query_params["recursive"] = "true"
+        if params is None:
+            params = {}
 
-    url = _helpers.update_query(base_url, query_params)
+        params["recursive"] = "true"
+
+    if params is not None:
+        url = _helpers.update_query(base_url, params)
+    else:
+        url = base_url
```

Comment:
This would be clearer if it just used `query_params` as before, defaulting it, e.g.
```python
    base_url = urlparse.urljoin(root, path)
    query_params = {} if params is None else params

    if recursive:
        query_params["recursive"] = "true"

    url = _helpers.update_query(base_url, params)
```